CMSIS/Core/Include/cmsis_compiler.h

```diff
@@ -53,6 +53,21 @@
  */
 #elif defined ( __ICCARM__ )
 
+  /* CMSIS compiler control architecture macros */
+  #if (__CORE__ == __ARM6M__) || (__CORE__ == __ARM6SM__)
```

Comment:
We should do some further clean up here. I suggest to:
- [ ] @JonatanAntoni check with IAR whether they plan to port this fix back to older release branches.
- [ ] @0xc0170 move the define block behind the include to cmsis_iar.h
- [ ] @0xc0170 check whether it is possible to exclude this block for compiler versions 8.10 and later. this would highlight the backward compatibility reason of those defines.